hertzbeat-alerter/src/main/java/org/apache/hertzbeat/alert/notice/AlertNoticeDispatch.java

```diff
@@ -121,14 +129,31 @@ public void dispatchAlarm(GroupAlert groupAlert) {
     }
 
     private void sendNotify(GroupAlert alert) {
-        matchNoticeRulesByAlert(alert).ifPresent(noticeRules -> noticeRules.forEach(rule -> workerPool.executeNotify(() -> rule.getReceiverId()
-                .forEach(receiverId -> {
-                    try {
-                        sendNoticeMsg(getOneReceiverById(receiverId),
-                                getOneTemplateById(rule.getTemplateId()), alert);
-                    } catch (AlertNoticeException e) {
-                        log.warn("DispatchTask sendNoticeMsg error, message: {}", e.getMessage());
-                    }
-                }))));
+        matchNoticeRulesByAlert(alert).ifPresent(noticeRules -> noticeRules.forEach(rule -> workerPool.executeNotify(() -> {
+            List<CompletableFuture<Void>> futures = rule.getReceiverId().stream()
+                    .map(receiverId -> sendNoticeAsync(getOneReceiverById(receiverId),
+                            getOneTemplateById(rule.getTemplateId()), alert))
+                    .toList();
+            
+            CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
+                    .whenComplete((result, exception) -> {
+                        if (exception != null) {
+                            log.warn("Some async notifications failed", exception);
+                        } else {
+                            log.debug("All notifications completed for alert: {}", alert.getGroupLabels());
+                        }
+                    });
+        })));
```

Comment:
hi can you describe why use the `CompleableFuture` here, `sendNoticeMsg` does not need to return a result; it just needs to handle exceptions.